### PR TITLE
Improve ubuntu-cloudimg template

### PR DESCRIPTION
**Pull request by troyanov**
This PR contains some refactoring that should simplify usage of this template and a bug fix related to `_VARS.fd` (that should be writable).

### CODE REVIEW

**Comment by bjornt** [comment]
+1

Nice trick to use a new build to get rid of the Makefile step.

Can you make sure that `make lint` still passes before merging?

**Comment by troyanov** [comment]
> Can you make sure that `make lint` still passes before merging?

Hmm, it was valid when I've tried it. Do you have any linter errors/warnings on your side?
```
packer validate -var customize_script=scripts/cloudimg/setup-boot.sh ubuntu-cloudimg.pkr.hcl
The configuration is valid.
packer fmt -check -diff .
```

**Comment by troyanov** [comment]
I'm actually wondering if we can just run it as:
`packer validate ubuntu-cloudimg.pkr.hcl` (without passing `customize_script`)

**Comment by troyanov** [comment]
@bjornt do you think if it is a good idea to add GitHub Action to run lint check as a part of CI?
Here is an example https://github.com/troyanov/packer-maas/commit/f6691d859e3b4a01b54f8ac5187dcec08ac78302

File: ubuntu/ubuntu-cloudimg.pkr.hcl [hunk 1]
```diff
@@ -128,6 +132,19 @@ source "qemu" "qemu" {
   use_backing_file       = true
 }
 
+build {
+  sources = ["source.null.dependencies"]
+
+  provisioner "shell-local" {
+    inline = [
+      "cp /usr/share/${lookup(local.uefi_imp, var.architecture, "")}/${lookup(local.uefi_imp, var.architecture, "")}_CODE.fd ${lookup(local.uefi_imp, var.architecture, "")}_CODE.fd",
```
Comment (alexsander-souza) [inline ubuntu/ubuntu-cloudimg.pkr.hcl:140]: not needed, but won't hurt

**Review by alexsander-souza — APPROVED** [review]
LGTM

just a nit

File: ubuntu/ubuntu-cloudimg.pkr.hcl (hunk 1 above)
Comment (troyanov) [reply ubuntu/ubuntu-cloudimg.pkr.hcl]: Yep, let's remove it not to confuse others of this not needed file copy.

**Review by alexsander-souza — APPROVED** [review]
+1

File: ubuntu/ubuntu-cloudimg.pkr.hcl [hunk 2]
```diff
@@ -114,8 +118,8 @@ source "qemu" "qemu" {
     ["-machine", "${lookup(local.qemu_machine, var.architecture, "")}"],
     ["-cpu", "${lookup(local.qemu_cpu, var.architecture, "")}"],
     ["-device", "virtio-gpu-pci"],
-    ["-drive", "if=pflash,format=raw,id=ovmf_code,readonly=on,file=/usr/share/${lookup(local.uefi_imp, var.architecture, "")}/${lookup(local.uefi_imp, var.architecture, "")}_CODE.fd"],
-    ["-drive", "if=pflash,format=raw,id=ovmf_vars,readonly=on,file=/usr/share/${lookup(local.uefi_imp, var.architecture, "")}/${lookup(local.uefi_imp, var.architecture, "")}_VARS.fd"],
+    ["-drive", "if=pflash,format=raw,id=ovmf_code,readonly=on,file=/usr/share/${lookup(local.uefi_imp, var.architecture, "")}_CODE.fd"],
```
[the remark this answers is not in the document]
Comment (troyanov) [reply ubuntu/ubuntu-cloudimg.pkr.hcl]: Yeah, I fixed it locally but didn't force push yet. Running a test build

**Pull request merged**